homeassistant/components/zwave_js/services.py

```diff
@@ -0,0 +1,175 @@
+
+    @callback
+    def async_register(self) -> None:
+        """Register all our services."""
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
+            self.async_set_config_parameter,
+            schema=vol.All(
+                {
+                    vol.Exclusive(ATTR_DEVICE_ID, "id"): cv.string,
+                    vol.Exclusive(ATTR_ENTITY_ID, "id"): cv.entity_ids,
+                    vol.Required(const.ATTR_CONFIG_PARAMETER): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                    vol.Optional(const.ATTR_CONFIG_PARAMETER_BITMASK): vol.Any(
+                        vol.Coerce(int), BITMASK_SCHEMA
+                    ),
+                    vol.Required(const.ATTR_CONFIG_VALUE): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                },
+                cv.has_at_least_one_key(ATTR_DEVICE_ID, ATTR_ENTITY_ID),
+                parameter_name_does_not_need_bitmask,
+            ),
+        )
+
+    @callback
+    def async_get_node_from_device_id(self, device_id: str) -> ZwaveNode:
+        """Get node from a device ID."""
+        device_entry = self._dev_reg.async_get(device_id)
+
+        if not device_entry:
+            raise ValueError("Device ID is not valid")
+
+        # Use device config entry ID's to validate that this is a valid zwave_js device
+        # and to get the client
+        config_entry_ids = device_entry.config_entries
+        config_entry_id = next(
+            (
+                config_entry_id
+                for config_entry_id in config_entry_ids
+                if cast(
+                    ConfigEntry,
+                    self._hass.config_entries.async_get_entry(config_entry_id),
+                ).domain
+                == const.DOMAIN
+            ),
+            None,
+        )
+        if (
+            config_entry_id is None
+            or config_entry_id not in self._hass.data[const.DOMAIN]
+        ):
+            raise ValueError("Device is not from an existing zwave_js config entry")
+
+        client = self._hass.data[const.DOMAIN][config_entry_id][const.DATA_CLIENT]
+
+        # Get node ID from device identifier, perform some validation, and then get the
+        # node
+        identifier = next(
+            (
+                get_home_and_node_id_from_device_id(identifier)
+                for identifier in device_entry.identifiers
+                if identifier[0] == const.DOMAIN
+            ),
+            None,
+        )
+
+        node_id = int(identifier[1]) if identifier is not None else None
+
+        if node_id is None or node_id not in client.driver.controller.nodes:
+            raise ValueError("Device node can't be found")
+
+        return client.driver.controller.nodes[node_id]
+
+    @callback
+    def async_get_node_from_entity_id(self, entity_id: str) -> ZwaveNode:
```

Comment:
same comment for this one. Might be nice to move to entity.py or a separate helpers.py